src/hotspot/share/opto/loopTransform.cpp

```diff
@@ -907,7 +909,18 @@ bool IdealLoopTree::policy_unroll(PhaseIdealLoop *phase) {
       future_unroll_cnt > LoopUnrollMin &&
       (future_unroll_cnt - 1) * (100.0 / LoopPercentProfileLimit) > cl->profile_trip_cnt() &&
       1.2 * cl->node_count_before_unroll() < (double)_body.size()) {
-    return false;
+    if (UseSuperWord && (cl->slp_max_unroll() == 0) &&
+        (cl->unrolled_count() - 1) * (100.0 / LoopPercentProfileLimit) <= cl->profile_trip_cnt()) {
```

Comment:
On line 911 and 913 this is repeated: 
"(X - 1) * (100.0 / LoopPercentProfileLimit) > cl->profile_trip_cnt()"
Please replace that with a method.